server_test.go

```diff
@@ -208,6 +208,51 @@ func TestOpenStatRace(t *testing.T) {
 	checkServerAllocator(t, server)
 }
 
+func TestOpenWithPermissions(t *testing.T) {
+	client, server := clientServerPair(t)
+	defer client.Close()
+	defer server.Close()
+
+	tmppath := path.Join(os.TempDir(), "open_permissions")
+	pflags := flags(os.O_RDWR | os.O_CREATE | os.O_TRUNC)
+	ch := make(chan result, 2)
```

Comment:
This channel should never have more than one in flight. (The design of `clientConn` guarantees an “at most once” delivery on a channel given to `dispatchRequest`, and after the first `dispatchRequest` this goroutine blocks waiting for that response.)